### DISCUSSION TITLE
How to setup Jest and handle "Unable to auto-detect a suitable renderer" error?

### DISCUSSION THREAD

**Discussion by mterczynski** [pixijs/pixijs#10119] [Q&A]
Hi, I want to add Jest to [my template project](https://github.com/mterczynski/pixijs-vite-template/) with PIXI and Vite.
I've added Jest with ts-jest, but when I run the tests I get the following error:

```
$ npm run test

> pixijs-template@1.0.0 test
> jest

 FAIL  src/index.test.ts
  ● Test suite failed to run

    Unable to auto-detect a suitable renderer.

      20 |
      21 | function initializeApp() {
    > 22 |      const app = new PIXI.Application({
         |                  ^
      23 |              width: settings.canvasWidth,
      24 |              height: settings.canvasHeight,
      25 |      });

      at Object.autoDetectRenderer (node_modules/@pixi/core/src/autoDetectRenderer.ts:67:11)
      at new autoDetectRenderer (node_modules/@pixi/app/src/Application.ts:107:25)
      at initializeApp (src/index.ts:22:14)
      at init (src/index.ts:5:14)
      at Object.<anonymous> (src/index.ts:34:1)
      at Object.<anonymous> (src/index.test.ts:1:1)
```
   
 I know that the main Pixi repository uses Jest, but I wasn't able to easily find how to copy some of its solutions to my project and make them work. I will appreciate any help.

**Comment by bigtimebuddy** [comment]
PixiJS uses [jest-electron](https://www.npmjs.com/package/jest-electron) runner with Jest to run within an actual browser context. Hope that helps.

**Reply by mterczynski** [reply]
I will check it, thanks!
